core/genesis.go

```diff
@@ -155,9 +155,12 @@ func (ga *GenesisAlloc) flush(db ethdb.Database) error {
 	if err != nil {
 		return err
 	}
-	err = statedb.Database().TrieDB().Commit(root, true, nil)
-	if err != nil {
-		return err
+	// Commit newly generated states into disk if it's not empty.
+	if root != types.EmptyRootHash {
+		err = triedb.Commit(root, true, nil)
+		if err != nil {
+			return err
+		}
```

Comment:
```suggestion
		if err := triedb.Commit(root, true, nil); err != nil {
			return err
		}
```
... ^ `=` or `:=` either works